make_queue/models/models.py

```diff
@@ -188,44 +186,44 @@ class Quota(models.Model):
-            ("can_edit_quota", "Can edit quotas"),
+            ('can_create_event_reservation', "Can create event reservation"),
+            ('can_edit_quota', "Can edit quotas"),
         )
 
     def get_active_reservations(self, user):
         if self.diminishing:
-            return self.reservation_set.all()
-        reservation_set = self.reservation_set if not self.all else self.reservation_set.filter(user=user)
-        return reservation_set.all().filter(end_time__gte=timezone.now())
+            return self.reservations.all()
+        reservations = self.reservations.filter(user=user) if self.all else self.reservations
+        return reservations.filter(end_time__gte=timezone.now())
 
     def can_create_more_reservations(self, user):
         return self.number_of_reservations != self.get_active_reservations(user).count()
 
-    def is_valid_in(self, reservation):
-        reservation_exists_or_can_make_more = (self.reservation_set.filter(pk=reservation.pk).exists()
+    def is_valid_in(self, reservation: 'Reservation'):
+        reservation_exists_or_can_make_more = (self.reservations.filter(pk=reservation.pk).exists()
                                                or self.can_create_more_reservations(reservation.user))
         ignore_rules_or_valid_time = (self.ignore_rules
                                       or ReservationRule.valid_time(reservation.start_time, reservation.end_time, reservation.machine.machine_type))
         return reservation_exists_or_can_make_more and ignore_rules_or_valid_time
 
-    @staticmethod
-    def can_make_new_reservation(user, machine_type):
-        return any(quota.can_create_more_reservations(user) for quota in Quota.get_user_quotas(user, machine_type))
+    @classmethod
+    def can_create_new_reservation(cls, user: User, machine_type: MachineType):
+        return any(quota.can_create_more_reservations(user) for quota in cls.get_user_quotas(user, machine_type))
 
     @staticmethod
-    def get_user_quotas(user, machine_type):
+    def get_user_quotas(user: User, machine_type: MachineType):
         return machine_type.quotas.filter(Q(user=user) | Q(all=True))
 
-    @staticmethod
-    def get_best_quota(reservation):
+    @classmethod
+    def get_best_quota(cls, reservation: 'Reservation'):
         """
         Selects the best quota for the given reservation,
-        by preferring non-diminishing quotas that do not ignore the rules
+        by preferring non-diminishing quotas that do not ignore the rules.
 
         :param reservation: The reservation to check
         :return: The best quota, that can handle the given reservation, or None if none can
         """
-        valid_quotas = [quota for quota in Quota.get_user_quotas(reservation.user, reservation.machine.machine_type) if
+        valid_quotas = [quota for quota in cls.get_user_quotas(reservation.user, reservation.machine.machine_type) if
                         quota.is_valid_in(reservation)]
```

Comment:
Maybe add cls in the documentation, so it can be clearer what it is.

Reply:
Hm, `cls` is the standard name for the class object in methods annotated with `@classmethod` (just like `self` is the standard name for the instance object in "normal" methods), and there are multiple hits that explain its purpose when googling "python cls", so it feels kind of redundant.. 🤔 I guess this boils down to where we should place a line for explaining features of the language / standard library when they are advanced or uncommon enough, but I personally feel like `@classmethod` and `cls` are such commonly seen features that they should not need to be explained. Any thoughts? 🙂

Reply:
I just had never seen it so I was a bit confused in the beginning. But after a quick search I found that you dont typically include `cls` or `self`, so I'll take back the request for changes.